src/DacpacTool/ModelValidationError.cs

```diff
@@ -9,34 +9,62 @@ namespace MSBuild.Sdk.SqlProj.DacpacTool
     /// </summary>
     public class ModelValidationError
     {
-        private readonly DacModelError _modelError;
         private readonly string _sourceName;
+        private readonly int _line;
+        private readonly int _column;
+        private readonly ModelErrorType _errorType;
+        private readonly ModelErrorSeverity _severity;
+        private readonly string _prefix;
+        private readonly int _errorCode;
+        private readonly string _message;
 
         public ModelValidationError(DacModelError modelError, string sourceName)
         {
-            _modelError = modelError ?? throw new ArgumentNullException(nameof(modelError));
-            _sourceName = sourceName.Replace("MSSQL::", string.Empty);
+            if (modelError == null) throw new ArgumentNullException(nameof(modelError));
```

Comment:
Perhaps this could be simplified to `modelError = modelError ?? throw new ArgumentNullException(nameof(modelError));`

Reply:
I don't follow why I would need to set `modelError = modelError`. Perhaps you mean `_modelError = modelError`, but I removed that internal `_modelError` field when I refactored this class.

Reply:
It is called a [throw expression](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/throw#the-throw-expression) but can only happen as part of an assignment so that's why you'd do the assignment. I know it is a bit weird.

I can live with the if as well, but then I'd rather have braces around the throw rather than the current one-liner.

Reply:
okay, got it, I'll make that change.